validationFunctions/validatePartnersAreDifferent.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Function for validating the partners array
+ * @module functions/validatePartnersAreDifferent
+ */
+
+/**
+ * Runs custom validation on the partners array. It will be true if the supplied argument is an array with no duplicate partner values.
+ *
+ * @param {Object[]} partners The list of partners supplied by the user
+ *
+ * @returns {boolean} It will return true if the argument is an array with no duplicate partner values, false otherwise.
+ */
+
+const validatePartnersAreDifferent = (
+  partners
+) => {
+  var i;
+  for (i = 1; i < partners.length; i++) {
+    if (
+      partners[i - 1].name === partners[i].name
```

Comment:
This only checks that 2 partners next to each other in the list are unique.  This may be the case if the list is ordered?  But would be safer to do a proper check of the whole list.  There is a simple way of doing this using a Set:
`var uniquePartners = new Set(partners.map(partner => partner.name));`
`return uniquePartners.size === partners.length;`